hell/entries/35-button-structure-by-table.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "#35 a perfectly structured button"
+date: 2025-09-11
+permalink: /{{ title | slug }}/index.html
+layout: layouts/entry.njk
+author: annedrtlf
+badcode: "<button type=\"button\"  onclick=\"window.open('https://example.com/other-page')\">
+<table><tbody><tr><td><i class=\"fa fa-arrow-left\" aria-hidden=\"true\"></i></td><td align=\"left\">Back</td></tr></tbody></table>
+</button>"
+goodcode: "<button type=\"button\" style=\"padding: 12px 16px;\" onclick=\"window.open('https://example.com/other-page')\"><i class=\"fa fa-arrow-left\" aria-hidden=\"true\"></i>Back</button>"
+---
+
+<div class="section bad">
+
+## Bad code
+
+```html
+{{ badcode | pretty }}
+```
+</div>
+<button type="button" onclick="window.open('example.com/other-page')">
+<table>
+<tbody>
+<tr>
+<td><i class="fa fa-arrow-left" aria-hidden="true"></i></td>
+<td align="left">Back</td>
+</tr>
+</tbody>
+</table>
+</button>
+<div class="section" id="issues">
+
+## Issues and how to fix them
+
+1. Technically it works to put a table inside a button. As the buttons content is representational the syntax is valid. Nevertheless it adds complexity to the code and is not needed.
```

Comment:
Do you mean presentational? If yes, please link the term. [Accessible Rich Internet Applications (WAI-ARIA) 1.2](https://www.w3.org/TR/wai-aria-1.2/#childrenArePresentational)

Here you say that the syntax is valid. Later you say the opposite.

I would split up your points differently. Here's a quick and dirty short version.

1. Technically it works but it's not valid. (link https://validator.w3.org/nu/?doc=https://cdpn.io/pen/debug/QwyyEzj) MDN etc.
2. Semantics of table not exposed because Button's content is presentational. (link to spec)
3. CSS
4. structure with phrasing content
5. Onclick on button -> you probably von <a href>